### PR TITLE
Rebase hoaprojecet/Database#18 on master

### PR DESCRIPTION
**Pull request by Hywan**
What do you think?

### CODE REVIEW

**Comment by Metalaka** [comment]
@Hywan Seem good, but why the PR target `master` and not `Metalaka:externalIterator` ?

File: Layer/Pdo/Iterator.php [hunk 1]
```diff
@@ -0,0 +1,193 @@
...
+    /**
+     * Create an iterator instance.
+     *
+     * @param   \PDOStatement  $statement      The PDOStatement instance.
+     * @param   array          $style          An array of fetching style
+     *                                         options.
+     * @return  void
+     */
+    public function __construct(
+        \PDOStatement $statement,
```
Comment (Metalaka) [inline Layer/Pdo/Iterator.php:83]: Should be `Hoa\Database\IDal\WrapperStaement` because of the interface.

Comment (Hywan) [inline Layer/Pdo/Iterator.php:83]: Yup, but it was in your original commit. I will let you edit this :-).

**Comment by Hywan** [comment]
Close in favor of #2.

File: Layer/Pdo/Iterator.php (hunk 1 above)
Comment (Metalaka) [inline Layer/Pdo/Iterator.php:83]: In my commit I wrote `\PDOStatement` on both interface and implementation ;)

Comment (Metalaka) [inline Layer/Pdo/Iterator.php:83]: I'll remove this type hint because there is no base between connections implementation.

**Pull request closed without merging**